compiler/AST/build.cpp

```diff
@@ -1862,13 +1862,19 @@ buildFunctionSymbol(FnSymbol*   fn,
 
   if (class_name)
   {
-    fn->_this = new ArgSymbol(thisTag,
-                              "this",
-                              dtUnknown,
-                              new UnresolvedSymExpr(class_name));
+    ArgSymbol* arg = new ArgSymbol(thisTag,
+				   "this",
+				   dtUnknown,
+				   new UnresolvedSymExpr(class_name));
+    fn->_this = arg;
+    if (thisTag == INTENT_TYPE) {
+      arg->intent = INTENT_BLANK;
+      arg->addFlag(FLAG_TYPE_VARIABLE);
```

Comment:
Since the "trick" in this patch is here -- allowing a type variable to bind to the "this" argument -- it would be good to put in a comment to that effect.
The code that allows this is elsewhere, not appearing in the changes included in this patch, so a pointer to that code would suffice -- especially if the implementation of <type>.<method_name> binding is described there.

Documentation about dependencies of this sort will be especially useful to Mike in his efforts to factor the resolution phase....